### PR TITLE
Add location tags and Castle Interior map

### PR DESCRIPTION
**Pull request by savage13**
- Add Location tags for items with Hyrule Castle
- Add a Hyrule Castle Interior map for reference (accessed from within the Filter pane).

As with the `locations` in the database, I am uncertain about what the names should be for the item locations (radar) and the Hyrule Interior polygons.

As usual, I am open to suggestions on improvements. 

I would love a polygon/model dump of the Castle, but I did the Hyrule Castle polygons by hand. :|

Screen Shot 2022-04-23 at 3 52 58 PM
 
Screenshot 2022-04-24 at 16-08-30 BotW Object Map

---
This change is [Reviewable](https://reviewable.io/reviews/zeldamods/objmap/84)

### CODE REVIEW

File: src/components/ObjectInfo.vue [hunk 1]
```diff
@@ -18,6 +18,10 @@
     </section>
     <section class="search-result-drop" v-if="!dropAsName && data.drop"><i class="fa fa-gem fa-fw"></i> {{drop()}}</section>
     <section class="search-result-equip" v-if="data.equip"><i class="fa fa-shield-alt fa-fw"></i> {{data.equip.map((x) => getName(x)).join(', ')}}</section>
+    <section class="search-result-rankup" v-if="data.location">
```
Comment (leoetlino) [inline src/components/ObjectInfo.vue:21]: search-result-location

File: src/components/ObjectInfo.vue [hunk 2]
```diff
@@ -18,6 +18,10 @@
...
+      <i class="fas fa-location-arrow" aria-hidden="true" style="color: honeydew"></i>
```
Comment (leoetlino) [inline src/components/ObjectInfo.vue:22]: `fa-fw` to ensure the icons are aligned

**Review by leoetlino — CHANGES_REQUESTED** [review]

**Review by savage13 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/zeldamods/objmap/84)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @leoetlino)

___
*[`src/components/ObjectInfo.vue` line 21 at r1](https://reviewable.io/reviews/zeldamods/objmap/84#-N0SLCdcQZry5HetS_Ul-r1-21:-N0z0ufKC6iFa84Ag6sf:b-896fix) ([raw file](https://github.com/zeldamods/objmap/blob/d2df98ed46613db98c0b4880771b87efce94a6ee/src/components/ObjectInfo.vue#L21)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary><blockquote>

search-result-location
</blockquote></details>

Done.
___
*[`src/components/ObjectInfo.vue` line 22 at r1](https://reviewable.io/reviews/zeldamods/objmap/84#-N0SLCdcQZry5HetS_Ul-r1-22:-N0z10be1LJWvnw9S_ut:b-896fix) ([raw file](https://github.com/zeldamods/objmap/blob/d2df98ed46613db98c0b4880771b87efce94a6ee/src/components/ObjectInfo.vue#L22)):*
<details><summary><i>Previously, leoetlino (Léo Lam) wrote…</i></summary><blockquote>

`fa-fw` to ensure the icons are aligned
</blockquote></details>

Done.

**Review by leoetlino — APPROVED** [review]

**Comment by leoetlino** [comment]
Oh, it looks like this fails to compile:

```
1065:54 Property 'y' does not exist on type 'Polygon | MultiPolygon'.
  Property 'y' does not exist on type 'Polygon'.
    1063 |     const entries = Object.entries(areas); // get the key later
    1064 |     for (const [data, features] of entries) {
  > 1065 |       let height = features.map((feature) => feature.y);
```

File: castle.json [hunk 3]
```diff
@@ -0,0 +1 @@
...70.6726139309194, -1020.6059658485553], [-269.77629491187116, -1020.2964456117616], [-268.8767517236896, -1020.0127187280341], [-267.97398436637485, -1019.7547851973727], [-267.0679928399268, -1019.5226450197774], [-266.15877714434544, -1019.3162981952485], [-265.24633727963084, -1019.1357447237855], [-264.330673245783, -1018.9809846053887], [-263.41178504280185, -1018.852017840058], [-262.48967267068747, -1018.7488444277934], [-261.5643361294398, -1018.671464368595], [-260.63577541905886, -1018.6198776624626], [-259.7039905395447, -1018.5844118019967], [-258.7689814908972, -1018.5650667871971], [-257.83074827311646, -1018.5618426180639], [-256.8892908862025, -1018.574739294597], [-255.94460933015523, -1018.6037568167964], [-254.99670360497464, -1018.648895184662], [-254.04557371066085, -1018.7101543981942], [-253.09121964721385, -1018.7875344573926], [-252.07238220110145, -1018.8939320387904], [-250.98906137232365, -1019.0293471423877], [-249.84125716088056, -1019.1937797681842], [-248.62896956677213, -1019.3872299161803], [-247.1684209494021, -1019.648387615975], [-245.4596113087705, -1019.977252867568], [-242.95120772308866, -1020.4898957597576]]], "color": "pink", "y": 192.0}]}
```
Comment (leoetlino) [inline castle.json:1]: @savage13 just noticed that the names do not match the JSON in the Radar repo. Could you provide an updated version of this? (and the generator script if you still have it – ideally it should be part of the repo)

Comment (savage13) [reply castle.json:1]: I did the digitization (polygons) and location naming by hand. :|  I was also wandering around in the game trying to find objects. This is the reason for the name mismatches and some missing polygons.

I should be able to construct a simple script that filters the the objects through the polygon file and assigns them locations based which region they are in.  

This should solve the mismatch between the polygon names and the radar locations.  

My apologies for the churn on this and thanks for you patience.

**Pull request merged**